apps/web/src/services/alert-manager/v2/composables/use-webhook-list-query.ts

```diff
@@ -0,0 +1,52 @@
+import type { Ref } from 'vue';
+import { computed } from 'vue';
+
+import type { QueryKey } from '@tanstack/vue-query';
+
+import { useWebhookApi } from '@/api-clients/alert-manager/webhook/composables/use-webhook-api';
+import type { WebhookModel } from '@/api-clients/alert-manager/webhook/schema/model';
+import { useScopedQuery } from '@/query/composables/use-scoped-query';
+import { useServiceQueryKey } from '@/query/query-key/use-service-query-key';
+
+import { useServiceDetailPageStore } from '../stores/service-detail-page-store';
+
+interface UseWebhookListQueryReturn {
+    webhookListData: Ref<WebhookModel[]>;
+    webhookListTotalCount: Ref<number>;
+    webhookListFetching: Ref<boolean>;
+    webhookListQueryKey: Ref<QueryKey>;
+}
+
+export const useWebhookListQuery = (): UseWebhookListQueryReturn => {
+    const serviceDetailPageStore = useServiceDetailPageStore();
+    const serviceDetailPageGetters = serviceDetailPageStore.getters;
+    const serviceId = computed<string>(() => serviceDetailPageGetters.serviceInfo.service_id);
```

Comment:
Suggestion: how about using `serviceId` as props of query composable?
I think, in this context, it would be better for responsibility of `serviceId` state to be managed at scenario level (component). Or at least, it would be better to handle `serviceId` state in `useScopedQuery's` `enabled`.